File: SupervisionRondier/agent.cpp
/**
  * \file agent.h
  * \brief Définition des méthodes de la classe Agent
  * \version 2.0
  * \author Gabriel Lemée
  * \date 2 avril 2017
  *
  * Classe modélisant un agent de sécurité.
  */

#include "agent.h"

/**
 * @brief Agent::Agent
 * Constructeur de la classe Agent
 * @param _nom Nom de l'agent
 * @param _prenom Prénom de l'agent
 * @param _badge Badge de l'agent
 * @param _agentActif Agent actif dans la base de données
 * @param _idAgent Id de l'agent dans la base de données
 */

Agent::Agent(QString _nom, QString _prenom, QString _badge, int _idAgent):
    nom(_nom),
    prenom(_prenom),
    badge(_badge),
    idAgent(_idAgent)
{

}

QString Agent::ObtenirNom() const
{
    return nom;
}

void Agent::DefinirNom(const QString &_nom)
{
    nom = _nom;
}

QString Agent::ObtenirPrenom() const
{
    return prenom;
}

void Agent::DefinirPrenom(const QString &_prenom)
{
    prenom = _prenom;
}

QString Agent::ObtenirBadge() const
{
    return badge;
}

void Agent::DefinirBadge(const QString &_badge)
{
    badge = _badge;
}

int Agent::ObtenirIdAgent() const
{
    return idAgent;
}

void Agent::DefinirIdAgent(int _idAgent)
{
    idAgent = _idAgent;
}

File: RondierProjet_Android_Source/Agent.java
package com.project.rondierprojet;

/**
 * @file Agent.java
 * @brief Définition de la classe de gestion d'un agent
 * @version 1.1
 * @author Tanguy Herbron
 * @date 3 Février 2017
 *
 * Classe modélisant un agent de sécurité
 */
public class Agent {

    private int id;
    private String idTag;
    private String nom;
    private String prenom;

    /**
     * @brief Constructeur Agent
     * Permet de créer un agent en fournissant son id, son nom ainsi que son prénom.
     * @param idTag Id correspondant au tag MiFare associé à l'agent.
     * @param nom Nom de l'agent.
     * @param prenom Prénom de l'agent.
     */
    public Agent(int id, String idTag, String nom, String prenom)
    {
        this.id = id;
        this.idTag = idTag;
        this.nom = nom;
        this.prenom = prenom;
    }

    /**
     * @brief Obtenir l'id d'un agent.
     * Permet d'obtenir l'id attribué à un agent.
     * @return Retourne l'id de l'agent.
     */
    public int obtenirId() {
        return id;
    }

    /**
     * @brief Obtenir le nom d'un agent.
     * Premet d'obtenir le nom attribué à un agent.
     * @return Le nom de l'agent.
     */
    public String obtenirNom()
    {
        return nom;
    }
}